registry/storage/manifeststore.go

```diff
@@ -13,6 +15,149 @@ import (
 	"github.com/docker/distribution/manifest/schema2"
 )
 
+const (
+	// MediaTypeOrphanedManifestRevision specifies the mediaType for orphaned
+	// (unlinked, delted) manifest that can be cleaned up. This type of
+	// manifest can be returned only during enumeration if enabled with option.
+	MediaTypeOrphanedManifestRevision = "application/vnd.docker.distribution.manifest.orphaned.v2+json"
```

Comment:
@stevvooe Please review [this helper](https://github.com/docker/distribution/pull/1050/files#diff-e566b0debd4b786531022c5420d4220aR85) for manifest service. It's used by [`manifestStore.Enumerate()`](https://github.com/docker/distribution/pull/1050/files#diff-e566b0debd4b786531022c5420d4220aR198). So there are 2 methods to do the same. The utility exported in `registry.storage` allows for better introspection.

If you get your blessing, I'll add some more unit tests.